### DISCUSSION TITLE
4.1.0rc2

### DISCUSSION THREAD

**Discussion by sadpandajoe** [apache/superset#29999] [General]
Hello Superset Community,

This is a call for the vote to release Apache Superset version 4.1.0.

The release candidate:
https://dist.apache.org/repos/dist/dev/superset/4.1.0rc2/

The Git tag for the release:
https://github.com/apache/superset/tree/4.1.0rc2

The CHANGELOG for the release:
https://github.com/apache/superset/blob/4.1.0rc2/CHANGELOG/4.1.0.md

The instructions for updating to the release:
https://github.com/apache/superset/blob/4.1.0rc1/UPDATING.md

Public keys are available at:
https://www.apache.org/dist/superset/KEYS

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/29997

**Reply by sadpandajoe** [reply]
Looks like this could be a blocker for people trying to upgrade, will cherry for rc3

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/29884

**Reply by sadpandajoe** [reply]
Not a blocker but will cherry as we'll need a rc3

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/pull/30017

if the sl_ tables don't exist, current migrations will fail

**Reply by sadpandajoe** [reply]
Could be a blocker to upgrade for users so will cherry it in once merged.

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30019

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30025

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30026

**Reply by sadpandajoe** [reply]
This creates logs but doesn't sound like it breaks functionality. Nice to get it in but doesn't seem like a blocker for 4.1.

**Reply by sadpandajoe** [reply]
fixed in: https://github.com/apache/superset/pull/30074

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30035

**Reply by sadpandajoe** [reply]
This seems like a blocker as the functionality is unusable if you're trying to drill on a lower dashboard.

**Reply by sadpandajoe** [reply]
fixed in: https://github.com/apache/superset/pull/30073

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by hao-zhuventures** [comment]
https://github.com/apache/superset/issues/30043

**Reply by sadpandajoe** [reply]
Here is a fix associated with the issue: https://github.com/apache/superset/pull/30042

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30054

**Reply by sadpandajoe** [reply]
doesn't feel like a blocker for 4.1, but let me see if I can find a root cause for this.

**Reply by sadpandajoe** [reply]
Looks like there's a fix here: https://github.com/apache/superset/pull/30077

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by padbk** [comment]
Probably not a blocker, but I'm seeing this issue https://github.com/apache/superset/issues/28833 when SLACK_ENABLE_AVATARS and websockets are enabled on 4.1.0rc2.

**Reply by sadpandajoe** [reply]
Agreed not a blocker for 4.1. Thanks @padbk for finding this

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/pull/30029

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/pull/30020

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/pull/29997

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30056

**Reply by sadpandajoe** [reply]
blocker for 4.1

**Reply by michael-s-molina** [reply]
@kgabryje @geido Can any of you help with this issue? I don't have much context about cross-filtering.

**Reply by sadpandajoe** [reply]
fix is here: https://github.com/apache/superset/pull/30176

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sfirke** [comment]
Not sure if this is worth an issue, but for both rc1 and rc2 my docker-compose deployment fails due to lack of psycopg2:
```
superset_app          | 2024-08-29 17:22:19,865:ERROR:superset.app:Failed to create app
superset_app          | Traceback (most recent call last):
superset_app          |   File "/app/superset/app.py", line 40, in create_app
superset_app          |     app_initializer.init_app()
superset_app          |   File "/app/superset/initialization/__init__.py", line 474, in init_app
superset_app          |     self.setup_db()
superset_app          |   File "/app/superset/initialization/__init__.py", line 637, in setup_db
superset_app          |     pessimistic_connection_handling(db.engine)
superset_app          |   File "/usr/local/lib/python3.10/site-packages/flask_sqlalchemy/__init__.py", line 998, in engine
superset_app          |     return self.get_engine()
superset_app          |   File "/usr/local/lib/python3.10/site-packages/flask_sqlalchemy/__init__.py", line 1017, in get_engine
superset_app          |     return connector.get_engine()
superset_app          |   File "/usr/local/lib/python3.10/site-packages/flask_sqlalchemy/__init__.py", line 594, in get_engine
superset_app          |     self._engine = rv = self._sa.create_engine(sa_url, options)
superset_app          |   File "/usr/local/lib/python3.10/site-packages/flask_sqlalchemy/__init__.py", line 1027, in create_engine
superset_app          |     return sqlalchemy.create_engine(sa_url, **engine_opts)
superset_app          |   File "<string>", line 2, in create_engine
superset_app          |   File "/usr/local/lib/python3.10/site-packages/sqlalchemy/util/deprecations.py", line 375, in warned
superset_app          |     return fn(*args, **kwargs)
superset_app          |   File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/create.py", line 544, in create_engine
superset_app          |     dbapi = dialect_cls.dbapi(**dbapi_args)
superset_app          |   File "/usr/local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 811, in dbapi
superset_app          |     import psycopg2
superset_app          | ModuleNotFoundError: No module named 'psycopg2'
```
I can start Superset if I add `psycopg2-binary==2.9.6` to my `requirements-local.txt` file.

My guess is that some part of the docker compose deployment process changed and I need to sync a newer version of files in the repo.  I'll look into that and I don't think this is a bug, but wanted to post here in case others were experiencing it.

**Reply by michael-s-molina** [reply]
@mistercrunch can you help with this one?

**Reply by mistercrunch** [reply]
on it!

**Reply by mistercrunch** [reply]
Is this a simple `docker-compose up` or `docker-compose -f docker-compose-non-dev.yml up` ? Will try and test both but could use more info to reproduce.

**Reply by mistercrunch** [reply]
I'd love to merge https://github.com/apache/superset/pull/30021 to save time while testing this... waiting for the frontend to build in-docker only to get flushed when starting the processes....

**Reply by mistercrunch** [reply]
Ok just tried rc2 (`git checkout 4.1.0rc2`) and ran:
1. `docker-compose build; docker-compose up` and couldn't reproduce (though I ran in some unrelated issues on `master`).
1.  Also ran `docker-compose -f docker-compose-non-dev.yml build; docker-compose -f docker-compose-non-dev.yml up` , worked too
1. Then tried `export TAG=4.1.0rc2; docker-compose -f docker-compose-image-tag.yml up` and hit your issue @sfirke , but realized that `4.1.0rc2` is the `lean` build, no database drivers included by design, so that you get the base and install just the drivers you need. If you use mysql you probably don't want the extra baggage. So you have to `export TAG=4.1.0rc2-dev; docker-compose -f docker-compose-image-tag.yml up` (notice the `-dev`). It's somewhat documented [here](https://superset.apache.org/docs/installation/docker-builds/#build-presets) but could be better. "lean builds don't contain any database driver, ..."

**Reply by mistercrunch** [reply]
here: https://github.com/apache/superset/pull/30144

**Reply by sfirke** [reply]
Thanks for digging into this, point 3 was indeed my case.  When I run `4.1.0rc2-dev` image I don't get this error.  This was a change from version 4.0.2 where I could run the lean image and it had the drivers, so great to get it cleared up in the docs.

Besides the psycopg2 driver, I was also getting a warning `INFO:superset.utils.screenshots:No PIL installation found` - adding that here for completeness in case it helps people stumble into the right place.  Looks like that would have affected screenshotting and probably only matters for the workers.  That warning is gone now too when running the `-dev` image.

**Reply by mistercrunch** [reply]
curious if/how we can consider this resolved. Do we delete the sub-thread?

**Reply by sadpandajoe** [reply]
we can just leave it open for anyone else running into issues but this isn't a blocker for 4.1

**Reply by sfirke** [reply]
We could add a note to the release notes and/or updating doc, pointing to the docker builds docs where @mistercrunch explained this and noting that as of 4.1.0 there are many drivers/packages that moved into the -dev image and are not present in lean?

(One more such package: I tried testing Excel uploads with the lean image and it failed because openpyxl was not installed. Works great with -dev)

**Reply by mistercrunch** [reply]
Not a blocker AFAIC, people should not use docker-compose to deploy in production. As far as ASF goes people should be using official tarballs. Adding a note in `UPDATING.md` about `lean` docker images being actually lean would be a nice thing to do though. I feel it's been the case for multiple releases though, no (?)

**Reply by sfirke** [reply]
Pretty sure it's new for 4.1.0, I'm running 4.0.2 and it came with `psycopg2-binary` etc.  I created a PR to add a note to the release notes about it: #30243

**Reply by sadpandajoe** [reply]
🍒 'd the note in the release notes

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30058

**Reply by rusackas** [reply]
I'll look into this one.

**Reply by justinpark** [reply]
Fixed in #30180

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30061

**Reply by michael-s-molina** [reply]
@kgabryje @geido Can any of you help with this issue? I don't have much context about cross-filtering.

**Reply by sadpandajoe** [reply]
looks like there is a fix here: https://github.com/apache/superset/pull/30252

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by Cmagno13** [comment]
The EVENT FLOW graph asks for a storm column but has no field to add.
Can you check?
[#30068](https://github.com/apache/superset/issues/30068)

<img width="880" alt="Captura de tela 2024-08-30 102833" src="https://github.com/user-attachments/assets/3bda9ca6-7b5e-4779-af24-35fb5c6ab386">

**Reply by sadpandajoe** [reply]
Looking at the issue, and it looks like it's also appearing in 4.0.2, so this won't be a blocker for 4.1's release.

**Reply by michael-s-molina** [reply]
@Cmagno13 keep in mind that there's also a [5.0 proposal](https://github.com/orgs/apache/projects/345/views/1?filterQuery=event&pane=issue&itemId=65364641) to remove this plugin.

**Reply by rusackas** [reply]
Indeed @Cmagno13 most of us on the thread are unlikely to turn our attention to this one due to the proposed deprecation. If you want to open a PR, we'll be happy to review it!

**Comment by sfirke** [comment]
- #30078

**Reply by sadpandajoe** [reply]
Probably not a blocker for 4.1, but we can see how long it'll take to fix this, if it's quick we can probably get it into rc3.

**Reply by michael-s-molina** [reply]
@sadpandajoe I believe this might be considered a blocker as it's a regression from 4.0.2.

**Reply by michael-s-molina** [reply]
fixed in: https://github.com/apache/superset/pull/30140

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sfirke** [comment]
- #30079

**Reply by michael-s-molina** [reply]
fixed in: https://github.com/apache/superset/pull/30138

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sfirke** [comment]
- #30136
Not a blocker IMO

**Reply by michael-s-molina** [reply]
fixed in: https://github.com/apache/superset/pull/30137

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/pull/30139 will fix previews in sql lab

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30064

**Reply by mistercrunch** [reply]
I vote for the solution being to:
- recommend read-uncommitted as the supported isolation level (more details in the issue)
- (?) curious what others think (?) demote `GLOBAL_ASYNC_QUERIES` as alpha / beta, meaning we wouldn't consider issues affecting it as release blocker

**Reply by sadpandajoe** [reply]
potential fix here: https://github.com/apache/superset/pull/30174

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30156

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30154

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by rabindragogoi** [comment]
@michael-s-molina : 
This issue (https://github.com/apache/superset/issues/30083 ) is fixed in 4.1.0rc2.

But hereI encountered another issue.
**_As an ADMIN user I cannot find the download csv option in the sql lab page_**.
Can you please help here?

![image](https://github.com/user-attachments/assets/a04fc0a4-0cc3-4e4a-8083-ccf68dfbce9e)




I cannot see the below download option in 4.1.0.rc2 which I can see in 3.0.0.

![image](https://github.com/user-attachments/assets/c9e39283-2d1b-4ec8-ba52-2ab03b8086d5)

**Reply by sadpandajoe** [reply]
@rabindragogoi can you create a new issue for this and we can link it to this discussion?

**Reply by rabindragogoi** [reply]
sure @sadpandajoe : New issue has been created: https://github.com/apache/superset/issues/30204

**Reply by sadpandajoe** [reply]
not a blocker for 4.1, see comments below

**Comment by lodu** [comment]
Would it be possible to make this part of the 4.1.0 release?
https://github.com/apache/superset/pull/28706

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by rabindragogoi** [comment]
@sadpandajoe , @michael-s-molina : Please find the link of  superset download to CSV issue: https://github.com/apache/superset/issues/30204

**Reply by sadpandajoe** [reply]
per the discussion in the issue, this appears in 4.0.2 for them too so won't be a blocker for 4.1

**Reply by sadpandajoe** [reply]
Upon further review, it looks like it is a regression in 4.1, so we'll need this fixed before we can move forward

**Reply by sadpandajoe** [reply]
from the issue, it looks like we couldn't repro this with 4.1 so the issue has been closed

**Comment by sfirke** [comment]
Could #30222 be cherried?

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30259

**Reply by justinpark** [reply]
Hotfix posted: https://github.com/apache/superset/pull/30272

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30260

**Reply by justinpark** [reply]
Hotfix posted: https://github.com/apache/superset/pull/30271

**Reply by sadpandajoe** [reply]
thanks @justinpark for the fix

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by round3d** [comment]
The table chart handling of both the new time comparisons and summaries is broken (incorrect results) when using custom filters. #30193

**Reply by sadpandajoe** [reply]
fixed in https://github.com/apache/superset/pull/30335

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/issues/30205

**Reply by sadpandajoe** [reply]
https://github.com/apache/superset/pull/30491

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30338

**Reply by michael-s-molina** [reply]
fixed in https://github.com/apache/superset/pull/30339

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/issues/30347

**Reply by sadpandajoe** [reply]
fixed with https://github.com/apache/superset/pull/30348

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by benvdh-incentro** [comment]
@sadpandajoe can you provide any estimate on when RC3 is going to be released? At our company we're eagerly awaiting the 4.1.0 release due to some features/bugfixes being present in that release. I've been following this thread for a while now (and the one on the dev list too), but so far I saw no one voting on the RC2 release on the mailing list (also no -1's), and only bugs + fixes being mentioned in this discussion that subsequently got cherry-picked into the release branch.

Do not get me wrong, I appreciate all the hard work going into a release, and understand it can take some time to get a release out, both in terms of the size of this project, and most OSS projects relying on volunteers (even though I am aware preset also plays a role here). I was mostly curious...

**Reply by sadpandajoe** [reply]
@benvdh-incentro as you noted that there haven't been any voting done on RC2 but the number of bugs that have been reported is currently not going to pass. I'm trying to minimize the number of RCs we're going to have to create so we're waiting for at least https://github.com/apache/superset/issues/30205 to be done before creating RC3 unless if any blockers are showing up in RC2.

**Reply by borismod** [reply]
Our company is also waiting for SuperSet release and it's a blocker. Snyk identified several vulnerabilities in SuperSet dependencies. There is a discussion to abandon SuperSet at the company level. May I suggest having a minor release that will contain only security fixes?

**Reply by sadpandajoe** [reply]
@borismod creating a superset release with just vulnerability fixes would still require a vote for it to go through so would take time to get an official release for that too. My suggestion is that you can either use the current RC2 as is knowing that these are the known bugs and they'll be fixed during the official release, or the other option is to branch off of 4.0.2 and cherry-pick they vulnerability fixes into that branch and use that until 4.1 is ready. I know it's taking awhile to get 4.1 out but as you can see from this discussion, there have been a lot of regressions that needed to be fixed.

**Reply by sfirke** [reply]
If anyone in the community wants to lead a 4.0.3 release, that could be a possibility. They could pick security/bugfix cherries onto 4.0.2. That's a matter of someone stepping forward who is willing and able to do the work.

**Reply by borismod** [reply]
According to the enterprise policy, we're not allowed to use RC versions. It actually puts us in a position that have to start looking for SuperSet alternatives.

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30389

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30393

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30405

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30416

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30061

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/issues/30424

**Reply by sadpandajoe** [reply]
fix here: https://github.com/apache/superset/pull/30425

**Comment by sadpandajoe** [comment]
https://github.com/apache/superset/pull/30258 and https://github.com/apache/superset/pull/30417

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30502

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/30551

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/30563

**Reply by sadpandajoe** [reply]
🍒 'd

**Comment by sadpandajoe** [comment]
Since RC2 failed, going to close this discussion until RC3 is created. To follow along blockers or to add new issues, feel free to follow the new project board here: https://github.com/orgs/apache/projects/418

**Comment by iercan** [comment]
This error still exist on our side. 
#29729
